xc/device.ml

```diff
@@ -1244,8 +1244,10 @@ let clean_shutdown (task: Xenops_task.t) ~xs (x: device) =
 			try (Xenctrl.domain_getinfo xc x.frontend.domid).Xenctrl.hvm_guest
 			with _ -> false
 			in
-		try release ~xc ~xs ~hvm devs x.frontend.domid x.frontend.devid
-		with _ -> ());
+		if hvm then 
```

Comment:
AFAICS, PCI.clean/hard_shutdown is not directly called by any PCI specific code path. However, the one occurrence I can find is via Domain.destroy -> Device.hard_shutdown -> PCI.hard_shutdown. So if we add the conditional branching here, we might have changed the semantics when we shutdown/destroy a PV domain. I'd agree with Rob's advice and keep it as it was, i.e. without the branching.  If we do so, the lines doing hvm testing is no longer needed, and the xc param is no longer needed.